docs/APP-CONNECTIVITY.md

```diff
@@ -0,0 +1,299 @@
+process. User is able to avoid EVE's DNS server from being used and provide its own list of DNS
+servers for applications.
+
+Additionally, DHCP server can be used to [propagate IP routes](#application-ip-routing)
+as well as IP addresses of NTP servers to applications. User is able to configure one NTP server
+IP per local network instance. This is then merged with NTP server(s) set for the NI uplink
+(received from external DHCP server or configured by the user), and propagated together
+to the application using the DHCP option 42 (56 in DHCPv6).
+
+DNS server which EVE provides for every Local NI allows resolution of application names
+to their IP addresses. This is very handy when applications need to talk to each other but
+cannot predict how EVE will allocate IP addresses between them.
+Every other name resolution request is just forwarded to DNS servers associated with
+the uplink (received from external DHCP server or configured by the user).
+
+Applications connected to local NI are also provided with access to Metadata HTTP server,
+running on local-only IP address 169.254.169.254. This can be used by applications
+to retrieve cloud-init configuration, obtain information from EVE (e.g. device UUID,
+hostname, uplink IP address) or to download [patch envelopes](PATCH-ENVELOPES.md).
+More information about metadata server can be found in [ECO-METADATA.md](ECO-METADATA.md).
+
+### Switch Network Instance
+
+Switch Network Instance is a simple L2-only bridge between connected applications and
+(optionally) an uplink network adapter. Traffic is only forwarded by the host network stack.
+This allows applications to directly access external endpoints and vice-versa.
+Switch network can be configured without uplink (i.e. as air-gapped), in which case it is merely
+a bridge between application VIFs.
+
+EVE does not run DHCP server for Switch NI. Instead, external DHCP server from the uplink's
+network can be configured by the user to provide IP addresses to applications. In case of air-gap
+switch NI, one of the applications can run DHCP server for the network.
+
+Application MAC addresses can be either statically user-configured, or generated by EVE.
+Since multiple edge devices running EVE can connect to the same network segment via switch NIs,
+the MAC address generator will apply a hash function to the app UUID in an attempt to produce
+MAC address unique across the entire fleet of devices (and not just within one device).
+
+Despite Switch NI operating only at the L2 layer, the L3/L4 [network ACLs](NETWORK-ACLS.md)
+are still supported and must be configured to allow anything beyond just DNS and DHCP to pass.
+Since there is no NAT between applications and external endpoints, properly configured
+inbound ACL rules are that much more important.
+
+Metadata HTTP server is run for switch network instance only if it has uplink port with
+an IP address.
+
+### Uplink Port
```

Comment:
I know we call it "uplink" in the API, but this is really the EVE management port(s).
I wonder if that terminology would be more clear throughout since e.g., eth3 could be pointing up towards the Internet but not be used for EVE management traffic. Thoughts?

Reply:
I agree that "management port" would be better. However, "uplink" is quite infiltrated now across the code base, in the API and in our (open and proprietary) documentation. I guess we are stuck with it now.